wordpress-mobile/WordPress-iOS
# Reader notifications

Second part of #8950 

This PR adds the possibility to switch on/off the site notifications within the Reader section. This action can be achieved using the Notice view (after a site is followed) or using the action sheet menu if the site is already followed.

![animation_1_resize](https://user-images.githubusercontent.com/912252/39236202-179db25a-4870-11e8-9c58-ef28bbca35bf.gif)![animation_2_resize](https://user-images.githubusercontent.com/912252/39236318-7113632a-4870-11e8-86f5-c93d6f3a3614.gif)

This PR includes different changes:
- There's a new DataModel version, 75 that includes 2 new models to store site and email notifications subscriptions: `ReaderSiteInfoSubscriptionEmail` and `ReaderSiteInfoSubscriptionPost`. They have a relationship with `ReaderSiteTopic`
- The `ReaderTopicService` has been modified and extended to communicate with the Remote Service and store the data
- `ReaderMenuViewController` has been modified to load the menu and the followed sites asynchronously using a dispatch group to notify when the operations are completed
- `ReaderStreamViewController` and `ReaderDetailViewController` have been modified to display the Notification alert and to include the notifications switch on/off within the action sheet menu

**To Test:**
- Install the app on an old version to make sure the migration to the new data model works
- Test the turn on/off site notifications actions within the reader stream (discover, followed sites, search...) and the post detail.
- Follow a site and turn on the site notifications using the Notice View action
- Follow/unfollow a site and turn on/off site notifications from the Action Sheet menu.
- Confirm that the turn on/off site notifications button appears within the Action Sheet menu only if a site is followed by the user.
